Bottom: fbc2d9f50bd02fc1ea2344c35a5fa2f10d8a8497
Top:    1e47401a69fc441705bd28b3d267ada3e6cefb0f
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-10-17 01:17:38 -0700

Refresh of inspect

---
diff --git a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
index a5e316f6fe9..7f460234ffb 100644
--- a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
+++ b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
@@ -315,6 +315,8 @@ protected:
   }
 
 #if PY_MAJOR_VERSION < 3
+  // The python 2 API declares some arguments as char* that should
+  // be const char *, but it doesn't actually modify them.
   static char *py2_const_cast(const char *s) { return const_cast<char *>(s); }
 #else
   static const char *py2_const_cast(const char *s) { return s; }
diff --git a/lldb/unittests/ScriptInterpreter/Python/PythonDataObjectsTests.cpp b/lldb/unittests/ScriptInterpreter/Python/PythonDataObjectsTests.cpp
index 7adfef664f1..3f0cb78b64c 100644
--- a/lldb/unittests/ScriptInterpreter/Python/PythonDataObjectsTests.cpp
+++ b/lldb/unittests/ScriptInterpreter/Python/PythonDataObjectsTests.cpp
@@ -694,11 +694,13 @@ TEST_F(PythonDataObjectsTest, TestCallable) {
   }
 
   {
-    const char *script = "class Foo: \n"
-                         "  def bar(self, x):\n"
-                         "     return x \n"
-                         "bar_bound   = Foo().bar \n"
-                         "bar_unbound = Foo.bar \n";
+    const char *script = R"(
+class Foo:
+  def bar(self, x):
+     return x
+bar_bound   = Foo().bar
+bar_unbound = Foo.bar
+)";
     PyObject *o =
         PyRun_String(script, Py_file_input, globals.get(), globals.get());
     ASSERT_FALSE(o == NULL);
